core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -185,16 +184,22 @@ public TableTestBase(int formatVersion) {
     this.V2Assert = new TableAssertions(2, formatVersion);
   }
 
-  @Before
+  public TableTestBase() {
```

Comment:
because there are so many subclasses of `TableTestBase`, we decided to introduce `TestBase`, which is a JUnit5 version of `TableTestBase`. That way, we don't have to adjust all subclasses in one go and can therefore split up the required changes into smaller chunks.

Reply:
As per my understanding, We won't be able to get rid of `TableTestBase` . `TestBase` has it's own implementation of `ParameterizedTestExtension`. Many of the subclass of `TableTestBase` has different flavour of `ParameterizedTestExtension`. I will plan to use `TestBase` wherever it fits. I will also add a javadoc for both class which one should be used when. Meanwhile , Could you please take a highlevel look for the Temp directory or file initialization? Also I will remove all the classes on which @tomtongue was working.

Reply:
@nk1506 `TestBase` is what needs to be used and it uses the `ParameterizedTestExtension` that is compatible with JUnit5. The `ParameterizedTestExtension` used by `TableTestBase` and its subclasses only works with JUnit4. If you look for example at #9790, you'll see how parameterized JUnit4 tests need to be migrated to parameterized JUnit5 tests.

That being said, every subclass of `TableTestBase` that is migrated to JUnit5 needs to now extend `TestBase` and use `org.apache.iceberg.ParameterizedTestExtension`

Reply:
@nastra what will you recommend for class like [TestAppenderFactory](https://github.com/apache/iceberg/blob/bcbcbb263ea7e13ab22d0feb918e207c7e42dbbd/data/src/test/java/org/apache/iceberg/io/TestAppenderFactory.java#L65) which has different set of parameters. 

```
@Parameterized.Parameters(name = "FileFormat={0}, Partitioned={1}")
  public static Object[] parameters() {
    return new Object[][] {
      new Object[] {"avro", false},
      new Object[] {"avro", true},
      new Object[] {"orc", false},
      new Object[] {"orc", true},
      new Object[] {"parquet", false},
      new Object[] {"parquet", true}
    };
  }
```

Reply:
this can be handled similar to how it's done for `TestFlinkMetaDataTable`